packages/iov-lisk/src/derivation.ts

```diff
@@ -25,3 +25,19 @@ export function pubkeyToAddress(pubkey: Uint8Array): Address {
   const addressString = Long.fromBytesLE(firstEightBytes, true).toString(10) + "L";
   return addressString as Address;
 }
+
+export function isValidAddress(address: string): boolean {
+  return isValidAddressWithEnding(address, "L");
+}
+
+export function isValidAddressWithEnding(address: string, ending: string): boolean {
+  const tail = address.slice(-ending.length);
+  if (tail !== ending) {
+    return false;
+  }
+  const addressString = address.slice(0, -ending.length);
+  // parse as unsigned base 10 number and verify re-encoding matches encoding
+  // this is no leading zeros, no decimals, within 0 <= addressString <= 2**64-1
+  const addressNumber = Long.fromString(addressString, true, 10); // true => unsigned
+  return addressNumber.toString(10) === addressString;
```

Comment:
Explicit uint64 range check is missing here.

Reply:
Long will only return up to 2**64 -1.
That means any number larger will be truncated somehow and return a different value.